app/src/main/java/rick_and_morty/data/event/UIState.kt

```diff
@@ -0,0 +1,10 @@
+package rick_and_morty.data.event
+
+import rick_and_morty.data.model.CharacterResultsDto
+
+data class UIState (
+    var isLoading : Boolean = false,
+    val isSuccess: List<CharacterResultsDto> = emptyList(),
+    val isFailure: Boolean = false,
+    val failureMessage: Exception? = null
```

Comment:
maybe `failure` only for the name?

also having `isFailure` + `failureMessage` leaves some room for inconsistent states e.g. `isFailure=true` + `failureMessage=null`. To keep them in sync we can do:

data class UiState(
    ......
    val failure: Exception? = null
) {
    val isFailure: Boolean = failure != null
}

Reply:
Done